### PR TITLE
netbsd: Adapt to BSD-common infra changes for FreeBSD

### PR DESCRIPTION
**Pull request by Ericson2314**
###### Description of changes

See https://github.com/NixOS/nixpkgs/pull/82131 for the rest of the changes for FreeBSD. This is PRed separately because it is a macOS moderate rebuild so we target staging.

The main change is that we CD to the path we're building *after* applying patches, so we can patch other parts of the tree (from `extraPaths`) as needed.

Another change is that `netbsd.install` no longer depends on `fts`, which it evidently no longer needs.

Also, rename `BSD_PATH` to `COMPONENT_PATH `.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 1]
```diff
@@ -297,13 +298,14 @@ in lib.makeScopeWithSplicing
       mandoc groff rsync
     ];
     skipIncludesPhase = true;
-    buildInputs = with self; compatIfNeeded ++ [ fts ];
```
Comment (Ericson2314) [inline pkgs/os-specific/bsd/netbsd/default.nix:300]: `fts` doesn't appear to be needed.

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 2]
```diff
@@ -281,11 +281,12 @@ in lib.makeScopeWithSplicing
 
   # HACK: to ensure parent directories exist. This emulates GNU
   # install’s -D option. No alternative seems to exist in BSD install.
-  install = let binstall = writeText "binstall" ''
-    #!${stdenv.shell}
-    for last in $@; do true; done
+  install = let binstall = writeScript "binstall" ''
+    #!${runtimeShell}
```
Comment (alyssais) [inline pkgs/os-specific/bsd/netbsd/default.nix:285]: Could just change to `writeShellScript` and make the shebang automatic if you want.

File: pkgs/os-specific/bsd/netbsd/default.nix [hunk 3]
```diff
@@ -281,11 +281,12 @@ in lib.makeScopeWithSplicing
...
+    set -eu
+    for last in "$@"; do true; done
     mkdir -p $(dirname $last)
-    xinstall "$@"
+    @out@/bin/xinstall "$@"
```
Comment (alyssais) [inline pkgs/os-specific/bsd/netbsd/default.nix:289]: Is this just because it's more correct, or did something break?

**Review by alyssais — COMMENTED** [review]
Approach makes sense.

So `BSD_PATH` is the path of the component we're building? Can we call it something less confusing?

File: pkgs/os-specific/bsd/netbsd/default.nix (hunk 2 above)
Comment (Ericson2314) [reply pkgs/os-specific/bsd/netbsd/default.nix:285]: good point!

**Comment by Ericson2314** [comment]
`BSD_PATH` was the name it was already using, but I am happy to change it.

**Comment by alyssais** [comment]
> `BSD_PATH` was the name it was already using, but I am happy to change it.

Yeah, I saw that you didn't introduce it.  I think it would be worth changing though, as I assumed it would refer to the root directory, which made seeing how the change worked quite confusing.

**Pull request merged**